### DISCUSSION TITLE
Having issue in IPFS cdn .

### DISCUSSION THREAD

**Discussion by sanu2002** [zkSync-Community-Hub/zksync-developers#813] [General] [answered]
### Environment

Testnet

### Provide a brief description of the functionality you're trying to implement and the issue you are running into.

I am facing the corresponding issue !

Uncaught SyntaxError: The requested module 'https://cdn.jsdelivr.net/npm/ipfs-core@0.17.0/dist/index.min.js' does not provide an export named 'default' (at Enterdata.html:11:20)
```
Here is my code
Enterdata.html

<!DOCTYPE html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="X-UA-Compatible" content="IE=edge">
        <title>Blockchain Form Submission</title>
        <meta name="description" content="">
        <meta name="viewport" content="width=device-width, initial-scale=1">
        <!-- Include the IPFS core library via CDN -->
        <script type="module">
            import {create}  from 'https://cdn.jsdelivr.net/npm/ipfs-core@0.17.0/dist/index.min.js';
        </script>
     
    </head>
    <style>
        h1 {
            font-size: 50px;
            color: blue;
            text-align: center;
            margin-top: 100px;
        }

        form {
            font-size: 50px;
            color: blue;
            text-align: center;
            margin-top: 100px;
        }
    </style>
    <body>
        <h1>Here we will store the data in IPFS & can be stored in Blockchain!</h1>
        <form id="form">
            <textarea id="textarea" name="textarea" style="height:200px; width:400px; border:2px solid red; box-shadow:2px 2px 2px"></textarea><br>
            <button type="submit" id="submit">Submit</button>
        </form>

        <script type="module" src="data.js"></script>
    </body>
</html>

```

```
const form = document.getElementById('form');
const textarea = document.getElementById('textarea');  // Get the textarea element
const submitButton = document.getElementById('submit');  // Get the submit button

form.addEventListener('submit', function (e) {
    e.preventDefault();  // Prevent the form from submitting and reloading the page

    const data = textarea.value;  // Get the value from the textarea

    console.log("Data submitted: " + data);  // Print to console

    // Create IPFS node and upload the data
    createNodeAndUploadData(data);
});

// Function to create IPFS node and upload the data
const form = document.getElementById('form');
const textarea = document.getElementById('textarea');  // Get the textarea element
const submitButton = document.getElementById('submit');  // Get the submit button

form.addEventListener('submit', function (e) {
    e.preventDefault();  // Prevent the form from submitting and reloading the page

    const data = textarea.value;  // Get the value from the textarea

    console.log("Data submitted: " + data);  // Print to console

    // Create IPFS node and upload the data
    createNodeAndUploadData(data);
});

// Function to create IPFS node and upload the data
async function createNodeAndUploadData(data) {
    try {
        // Create an IPFS node
        const ipfs = await create();
        console.log('IPFS node created:', ipfs.id);

        // Upload the data to IPFS
        const file = await ipfs.add(data);
        console.log("File uploaded to IPFS with hash:", file.path);

        // After getting the IPFS hash, you can store it in the blockchain
        // Example: storeHashInBlockchain(file.path);
    } catch (error) {
        console.error("Error with IPFS operations:", error);
    }
}

```

**Comment by alhenaai** [comment]
It seems like you're encountering an issue with importing the IPFS library in your HTML file. The error message indicates that the module you're trying to import does not have a default export named 'create'. This is a common issue when the module's exports do not match the import statements in your code.

To resolve this issue, you should check the documentation for the specific version of the IPFS library you are using (`ipfs-core@0.17.0`). Ensure that you are using the correct import syntax according to the library's export structure. If the library has named exports instead of a default export, you will need to adjust your import statement accordingly.

For example, if the library exports `create` as a named export, your import statement seems correct. However, if the error persists, it might be helpful to look at the library's documentation or repository to find the correct way to import and use the functions you need.

Your feedback helps us serve you better. If this fully addresses your question, please give it a thumbs up. If it doesn't, please give it a thumbs down. This will inform the community and team so we can improve our support.

**Reply by sanu2002** [reply]
yes sure !!

**Reply by alhenaai** [reply]
If you continue to experience issues, it might also be helpful to check if there are any updates or changes in the newer versions of the library that might affect how imports are handled. Sometimes, upgrading to a newer version can resolve these types of issues if they stem from bugs or deprecated features in older versions.

If you need further assistance, feel free to ask!

Your feedback helps us serve you better. If this fully addresses your question, please give it a thumbs up. If it doesn't, please give it a thumbs down. This will inform the community and team so we can improve our support.

**Reply by sanu2002** [reply]
in mynode env everything is working fine but wen i am working browser this type things happening

**Reply by alhenaai** [reply]
When working with Node.js environments versus browsers, there are often differences in how modules are handled due to the environments' distinct module resolution systems. In Node.js, you might be using CommonJS modules, which handle `require()` and `module.exports` differently from ES Modules, which use `import` and `export` and are typically used in browsers.

For your browser issue, it seems like the module you are trying to import does not have the expected export when loaded directly from the CDN in a browser environment. Here are a few steps you can take to resolve this:

1. **Verify the Module's Browser Support**: Check the documentation for `ipfs-core` to ensure that the version you are using supports direct import in browser environments via a CDN. Not all modules or their versions support this, and some might require specific builds or additional configuration.

2. **Use a Correct Import Statement**: Ensure that your import statement matches the module's documentation for browser usage. Sometimes, the path or the export name might differ slightly for browser environments.

3. **Consider Using a Bundler**: Tools like Webpack, Rollup, or Parcel can help bundle your dependencies into a single file that handles module resolution more gracefully across environments. This approach can also help with issues related to different module formats (like CommonJS vs. ES Modules).

4. **Check for CORS Issues**: Sometimes, loading modules from a CDN can lead to Cross-Origin Resource Sharing (CORS) issues. Ensure that the CDN headers are compatible with your usage in a browser context.

5. **Fallback to a Different Version or Build**: If the specific version of `ipfs-core` you are using does not work as expected in the browser, consider trying a different version or a build that is specifically meant for browser use.

If you need specific guidance on configuring your project with a bundler or adjusting your import statements according to `ipfs-core` documentation, feel free to ask for more detailed instructions.

Your feedback helps us serve you better. If this fully addresses your question, please give it a thumbs up. If it doesn't, please give it a thumbs down. This will inform the community and team so we can improve our support.

**Reply by sanu2002** [reply, marked as answer]
This package has been deprecated!!

js-IPFS has been deprecated in favour of Helia - please see https://github.com/ipfs/js-ipfs/issues/4336 for details

This is the message from the author

**Reply by alhenaai** [reply]
Given that the `js-ipfs` package has been deprecated in favor of Helia, you will need to migrate your project to use the new Helia library. This transition is necessary to ensure continued support and updates, as well as to take advantage of any new features and improvements offered by Helia.

Here are the steps you should consider for migrating to Helia:

1. **Visit the Helia Repository**: Start by visiting the [Helia GitHub repository](https://github.com/ipfs/js-ipfs/issues/4336) to understand the new library's structure, features, and usage guidelines.

2. **Install Helia**: Remove the deprecated `js-ipfs` package and install Helia. You can typically do this via npm or yarn. Check the repository for the exact installation command.

3. **Update Your Import Statements**: Replace the import statements in your code from `ipfs-core` to the appropriate Helia imports. The exact syntax and module structure might differ, so refer to the Helia documentation for details.

4. **Refactor Your Code**: Adjust your code to use Helia's API. This might involve changing function calls, handling promises or async functions differently, and updating configurations.

5. **Test Thoroughly**: After making these changes, thoroughly test your application to ensure that all functionalities work as expected without errors. Pay special attention to the parts of your application that interact with IPFS.

6. **Seek Help if Needed**: If you encounter issues or if the migration process is unclear, consider seeking help from the Helia community or documentation. They might provide migration guides or support forums where you can ask for assistance.

By following these steps, you should be able to successfully transition your project from the deprecated `js-ipfs` library to the new Helia library. This will help maintain the robustness and efficiency of your application in handling IPFS operations.

Your feedback helps us serve you better. If this fully addresses your question, please give it a thumbs up. If it doesn't, please give it a thumbs down. This will inform the community and team so we can improve our support.

**Reply by sanu2002** [reply]
Now it's fine with helia , thanks !!

**Reply by alhenaai** [reply]
You're welcome! I'm glad to hear that everything is working fine with Helia. If you have any more questions or need further assistance in the future, feel free to reach out. Happy coding!

Your feedback helps us serve you better. If this fully addresses your question, please give it a thumbs up. If it doesn't, please give it a thumbs down. This will inform the community and team so we can improve our support.
